auth/user_mgt_test.go

```diff
@@ -0,0 +1,704 @@
+				PhotoURL:    "http://www.example.com/testuser/photo.png",
+				Email:       "testuser@example.com",
+			}, {
+				ProviderID:  "phone",
+				PhoneNumber: "+1234567890",
+			},
+		}},
+		{user.DisplayName, "Test User"},
+		//		{user.PasswordHash, "passwordhash"},
+		//		{user.PasswordSalt, "salt==="},
+		{user.CustomClaims, map[string]interface{}{"admin": true, "package": "gold"}},
+	}
+	for _, test := range tests {
+		if !reflect.DeepEqual(test.want, test.got) {
+			t.Errorf("got %#v wanted %#v", test.got, test.want)
+		}
+	}
+}
+
+var listUsers []*ExportedUserRecord
+
+func setListUsers() {
+	listUsers = []*ExportedUserRecord{
+		{
+			UserRecord: &UserRecord{
+				UserInfo: &UserInfo{
+					UID: "VHHROt3NAjPoc1hanwMRcTdCESz2",
+				},
+				UserMetadata: &UserMetadata{
+					LastLogInTimestamp: 0,
+					CreationTimestamp:  1511284665000,
+				},
+				Disabled: false,
+			},
+		},
+		{
+			UserRecord: &UserRecord{
+				UserInfo: &UserInfo{
+					UID:         "tefwfd1234",
+					DisplayName: "display_name",
+					Email:       "tefwfd1234eml5f@test.com",
+				},
+				UserMetadata: &UserMetadata{
+					LastLogInTimestamp: 0,
+					CreationTimestamp:  1511284665000,
+				},
+				Disabled:      false,
+				EmailVerified: false,
+				ProviderUserInfo: []*UserInfo{
+					{
+						ProviderID:  "password",
+						DisplayName: "display_name",
+						Email:       "tefwfd1234eml5f@test.com",
+					},
+				},
+				CustomClaims: map[string]interface{}{"asssssdf": true, "asssssdfdf": "ffd"},
+			},
+			PasswordHash: "V4X0yt9qGyp6cfw6BNwRHdS4SDwgTKtUSZcW2LEBFRuadpYJePqOsHyNtEszBaO3veC_6eA24PF06gH61Ghq8w==",
+			PasswordSalt: "BxzGq0di67rcTw==",
+		},
+		{
+			UserRecord: &UserRecord{
+				UserInfo: &UserInfo{
+					DisplayName: "Test User",
+					Email:       "testuser@example.com",
+					UID:         "testuser0",
+					PhoneNumber: "+1234567890",
+					PhotoURL:    "http://www.example.com/testuser/photo.png",
+				},
+				UserMetadata: &UserMetadata{
+					LastLogInTimestamp: 0,
+					CreationTimestamp:  1234567890,
+				},
+				Disabled:      false,
+				EmailVerified: true,
+				ProviderUserInfo: []*UserInfo{
+					{
+						ProviderID:  "password",
+						DisplayName: "Test User",
+						Email:       "testuser@example.com",
+						PhotoURL:    "http://www.example.com/testuser/photo.png",
+					},
+					{
+						ProviderID:  "phone",
+						PhoneNumber: "+1234567890",
+					},
+				},
+				CustomClaims: map[string]interface{}{"admin": true, "package": "gold"},
+			},
+			PasswordHash: "passwordHash",
+			PasswordSalt: "passwordSalt",
+		},
+	}
+}
+
+type basicCompare struct {
+	got  interface{}
+	want interface{}
+}
+
+func TestGetUsers(t *testing.T) {
+	setListUsers()
+	s, closer := echoServer("list_users.json", t)
+	defer closer()
+	iter := s.Client().Users(context.Background(), "")
+	i := 0
+	for {
+		user, err := iter.Next()
+		if err == iterator.Done {
+			break
+		}
+		if err != nil {
+			t.Error(err)
+
+		}
+		tests := []basicCompare{
```

Comment:
Same comment as earlier. If you make a single reusalbe function to compare UserRecord instances, you can just call it here.